build.gradle

```diff
@@ -753,12 +753,19 @@ project(':clients') {
   task determineCommitId {
     ext.commitId = "unknown"
     def takeFromHash = 16
-    if (file("$rootDir/.git/HEAD").exists()) {
+
+    // Allow build system to set the commit ID as .git/HEAD might not be correct
+    // if there are local commits added for build purposes.
+    def kafkaCommitId = System.getenv('KAFKA_COMMIT_ID')
```

Comment:
That will require more changes on build side, but yes, consistency is good. I will make the change and also update README. Thx!